### PR TITLE
Support --with-pg_query-workdir to specific workdir

### PR DESCRIPTION
**Pull request by xiaohui-zhangxh**
For some reason download `libpg_query-xx.tar.gz` from https://codeload.github.com/lfittl/libpg_query/tar.gz/ is very slow(Download from China),
we can pre-download this file to `/tmp/`, then use `--with-pg_query-workdir=/tmp/` to compile it without downloading.

```
gem install pg_query --with-pg_query-workdir=/tmp/
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,13 @@ gem install pg_query
 
 Due to compiling parts of PostgreSQL, installation might take a while on slower systems. Expect up to 5 minutes.
 
+For some reason download `libpg_query-xx.tar.gz` from https://codeload.github.com/lfittl/libpg_query/tar.gz/ is very slow(Download from China),
```
Comment (olleolleolle) [inline README.md:21]:
```suggestion
For some reason download `libpg_query-xx.tar.gz` from https://codeload.github.com/lfittl/libpg_query/tar.gz/ is very slow when downloading from China.
```

File: README.md [hunk 2]
```diff
@@ -18,6 +18,13 @@ gem install pg_query
...
+we can pre-download this file to `/tmp/`, then use `--with-pg_query-workdir=/tmp/` to compile it without downloading.
```
Comment (olleolleolle) [inline README.md:22]:
```suggestion
We can pre-download this file to `/tmp/`, then use `--with-pg_query-workdir=/tmp/` to compile it without downloading.
```

**Review by olleolleolle — COMMENTED** [review]
Cool addition!

**Comment by lfittl** [comment]
@xiaohui-zhangxh Thanks for the contribution!

I solved this in a different way (see https://github.com/pganalyze/pg_query/commit/e271d8cebfc8b8ac99359c48da54d1f09d750ef3), because the whole code download on build is a bit problematic anyway.

In the new release (2.0.0), the gem itself includes the full source code, so I believe this should resolve the issue.

Could you confirm whether that resolves it for you?

**Comment by xiaohui-zhangxh** [comment]
> @xiaohui-zhangxh Thanks for the contribution!
> 
> I solved this in a different way (see [e271d8c](https://github.com/pganalyze/pg_query/commit/e271d8cebfc8b8ac99359c48da54d1f09d750ef3)), because the whole code download on build is a bit problematic anyway.
> 
> In the new release (2.0.0), the gem itself includes the full source code, so I believe this should resolve the issue.
> 
> Could you confirm whether that resolves it for you?

This would be good with neither additional network fetching nor building arguments. 👍

**Pull request closed without merging**